sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -260,7 +270,7 @@ abstract class StateStoreChangelogReader(
   }
   protected val input: DataInputStream = decompressStream(sourceStream)
 
-  def close(): Unit = { if (input != null) input.close() }
+  override protected def close(): Unit = { if (input != null) input.close() }
```

Comment:
Make sure `StateStoreChangelogReader` correctly implements the interface of `NextIterator`, method `closeIfNeeded` should be used instead of `close`.